File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnTertiaryFixedLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnTertiaryFixedLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl39
            AppTheme.STANDARD -> R.color.cstl39
            AppTheme.BLUE -> R.color.cbl39
            AppTheme.GREEN -> R.color.cgl39
            AppTheme.RED -> R.color.crl39
            AppTheme.NOTHING -> R.color.cnol39
        }
    }
}

File: lib/layer2_core/light/e_on_tertiary_fixed_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnTertiaryFixedLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl39;
      case 'standard': return cstl39;
      case 'blue': return cbl39;
      case 'green': return cgl39;
      case 'red': return crl39;
      case 'nothing': return cnol39;
      default: return cstl39;
    }
  }
}
